src/Evolution/Systems/NewtonianEuler/BoundaryConditions/Outflow.cpp

```diff
@@ -0,0 +1,122 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/NewtonianEuler/BoundaryConditions/Outflow.hpp"
+
+#include <cstddef>
+#include <memory>
+#include <optional>
+#include <pup.h>
+#include <string>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tags/TempTensor.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/BoundaryConditions/BoundaryCondition.hpp"
+#include "Evolution/Systems/NewtonianEuler/SoundSpeedSquared.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeString.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace NewtonianEuler::BoundaryConditions {
+template <size_t Dim>
+Outflow<Dim>::Outflow(CkMigrateMessage* const msg)
+    : BoundaryCondition<Dim>(msg) {}
+
+template <size_t Dim>
+std::unique_ptr<domain::BoundaryConditions::BoundaryCondition>
+Outflow<Dim>::get_clone() const {
+  return std::make_unique<Outflow>(*this);
+}
+
+template <size_t Dim>
+void Outflow<Dim>::pup(PUP::er& p) {
+  BoundaryCondition<Dim>::pup(p);
+}
+
+template <size_t Dim>
+// NOLINTNEXTLINE
+PUP::able::PUP_ID Outflow<Dim>::my_PUP_ID = 0;
+
+template <size_t Dim>
+template <size_t ThermodynamicDim>
+std::optional<std::string> Outflow<Dim>::dg_outflow(
+    const std::optional<tnsr::I<DataVector, Dim, Frame::Inertial>>&
+        face_mesh_velocity,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>&
+        outward_directed_normal_covector,
+
+    const Scalar<DataVector>& mass_density,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& velocity,
+    const Scalar<DataVector>& specific_internal_energy,
+    const EquationsOfState::EquationOfState<false, ThermodynamicDim>&
+        equation_of_state) {
+  double min_char_speed = std::numeric_limits<double>::signaling_NaN();
+
+  // Compute sound speed
+  Scalar<DataVector> sound_speed{get(mass_density).size()};
```

Comment:
why not do this allocation within the `Variables` a few lines down?

Reply:
Included `sound_speed` allocation into it.